src/event-replay/helpers.ts

```diff
@@ -26,6 +35,56 @@ export async function findTsvBlockHeight(filePath: string): Promise<number> {
   return blockHeight;
 }
 
+/**
+ * Traverse a TSV file to find the genesis block and extract its data so we can use it during V1 BNS
+ * import.
+ * @param filePath - TSV path
+ * @returns Genesis block data
+ */
+export async function findBnsGenesisBlockData(filePath: string): Promise<BnsGenesisBlock> {
+  const rl = readline.createInterface({
+    input: fs.createReadStream(filePath),
+    crlfDelay: Infinity,
+  });
+  let genesisBlock: BnsGenesisBlock | undefined;
+  return new Promise<BnsGenesisBlock>((resolve, reject) => {
+    rl.on('line', line => {
```

Comment:
How's memory usage with this? It's probably alright since there's no async code in the callback, if not, safer to handle stream backpressure, e.g. with an async iterator `for await (const line of rl) { ... }`